### DISCUSSION TITLE
adjust max machines dynamically

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#50] [Q&A] [answered]
I'm planning for my users to be able to adjust the max number of machines in their subscriptions. On the keygen side, is it possible to adjust the max number of machines in the respective license somehow?

Since _max machines_ is set in the policy, I guess I would only be able to offer a fixed number of _max machines_ levels (say 1, 5, 20, and 100), and upon subscription change adapt the policy of the respective license.

Is there way to freely adjust _max machines_?

**Comment by ezekg** [comment, marked as answer]
Using the API, you can [set a max machines override per-license](https://keygen.sh/docs/api/licenses/#licenses-update-attrs-maxMachines). Right now, this is an API-only feature.

Sounds like that should work for you here?

**Reply by nschloe** [reply]
Yup, that works. Thanks!
